src/components/application_manager/rpc_plugins/sdl_rpc_plugin/test/commands/hmi/subscribe_button_request_test.cc

```diff
@@ -0,0 +1,141 @@
+
+#include "gtest/gtest.h"
+
+namespace test {
+namespace components {
+namespace commands_test {
+namespace hmi_commands_test {
+namespace subscribe_button_request {
+
+using ::testing::_;
+using ::testing::Return;
+namespace am = ::application_manager;
+namespace strings = am::strings;
+using am::commands::CommandImpl;
+using am::commands::RequestToHMI;
+using am::event_engine::Event;
+using sdl_rpc_plugin::commands::hmi::SubscribeButtonRequest;
+using ::test::components::application_manager_test::MockApplication;
+
+typedef std::shared_ptr<MockApplication> MockAppPtr;
+typedef std::shared_ptr<RequestToHMI> RequestToHMIPtr;
+typedef std::shared_ptr<SubscribeButtonRequest> SubscribeButtonRequestPtr;
+
+namespace {
+const uint32_t kCorrelationId = 2u;
+const uint32_t kAppId = 1u;
+const hmi_apis::Common_ButtonName::eType kButtonName =
+    hmi_apis::Common_ButtonName::CUSTOM_BUTTON;
+const hmi_apis::FunctionID::eType kFunctionID =
+    hmi_apis::FunctionID::Buttons_SubscribeButton;
+}  // namespace
+
+class HMISubscribeButtonRequestTest
+    : public CommandRequestTest<CommandsTestMocks::kIsNice> {
+ protected:
+  MessageSharedPtr CreateCommandMsg() {
+    MessageSharedPtr command_msg(CreateMessage(smart_objects::SmartType_Map));
+    (*command_msg)[strings::msg_params][strings::app_id] = kAppId;
+    (*command_msg)[strings::msg_params][strings::button_name] = kButtonName;
+    (*command_msg)[strings::params][strings::correlation_id] = kCorrelationId;
+    (*command_msg)[strings::params][strings::function_id] = kFunctionID;
+
+    return command_msg;
+  }
+
+  void InitCommand(const uint32_t& timeout) OVERRIDE {
+    mock_app = CreateMockApp();
+    CommandRequestTest<CommandsTestMocks::kIsNice>::InitCommand(timeout);
+    ON_CALL((*mock_app), hmi_app_id()).WillByDefault(Return(kAppId));
+    ON_CALL(app_mngr_, application_by_hmi_app(kAppId))
+        .WillByDefault(Return(mock_app));
+    ON_CALL(app_mngr_, application(kAppId)).WillByDefault(Return(mock_app));
+  }
+
+  MockAppPtr mock_app;
+};
+
+TEST_F(HMISubscribeButtonRequestTest, Run_SendRequest) {
+  MessageSharedPtr command_msg = CreateCommandMsg();
+  RequestToHMIPtr command(CreateCommand<SubscribeButtonRequest>(command_msg));
+
+  EXPECT_CALL(mock_rpc_service_, SendMessageToHMI(command_msg));
+  ASSERT_TRUE(command->Init());
+
+  command->Run();
+
+  EXPECT_EQ(CommandImpl::hmi_protocol_type_,
+            (*command_msg)[strings::params][strings::protocol_type].asInt());
+  EXPECT_EQ(CommandImpl::protocol_version_,
+            (*command_msg)[strings::params][strings::protocol_version].asInt());
+}
+
+TEST_F(HMISubscribeButtonRequestTest, onTimeOut) {
+  MessageSharedPtr command_msg = CreateCommandMsg();
+  RequestToHMIPtr command(CreateCommand<SubscribeButtonRequest>(command_msg));
+
+  resumption_test::MockResumeCtrl mock_resume_ctrl;
+  EXPECT_CALL(app_mngr_, resume_controller())
+      .WillOnce(ReturnRef(mock_resume_ctrl));
+  EXPECT_CALL(mock_resume_ctrl, HandleOnTimeOut(_, _));
```

Comment:
Try to avoid underscores when you know which values to expect
```suggestion
  EXPECT_CALL(mock_resume_ctrl, HandleOnTimeOut(kCorrelationId, kFunctionID));
```